perception_lm: remove perception_encoder from auto-maps

omsherikar opened the pull request:
### What does this PR do?

Removes references to a non-existent base model from the Auto mappings so enumeration and lazy imports work as expected.

- Remove `perception_encoder` entries from Auto mappings
  - `src/transformers/models/auto/modeling_auto.py`
  - `src/transformers/models/auto/configuration_auto.py`
- Keep existing `perception_lm` entries intact
- Regenerate modeling files (no runtime behavior change)

Fixes: [#41387](https://github.com/huggingface/transformers/issues/41387)

### Motivation

`MODEL_MAPPING.items()` failed because Auto mappings referenced `perception_encoder`, which doesn’t exist. Per maintainer guidance, we should remove non-existent models from auto-map rather than adding aliases.

### Dependencies

None.

### Before submitting

- [x] Minimal change; no behavior changes.
- [x] Linked the issue.
- [x] No docs changes required.
- [x] Local check with `PYTHONPATH=src`:
```python
from transformers.models.auto.modeling_auto import MODEL_MAPPING
list(MODEL_MAPPING.items())  # succeeds
```

### Who can review?

- Model loading / auto mappings: @CyrilVallez
- Multimodal: @zucchini-nlp

omsherikar committed:
perception_lm: export PerceptionEncoder alias for auto mapping

omsherikar committed:
Merge pull request #1 from omsherikar/fix/perceptionlm-auto-alias

perception_lm: export PerceptionEncoder alias for auto mapping

omsherikar committed:
Merge branch 'main' into main

omsherikar committed:
Merge branch 'main' into main

zucchini-nlp commented:
I just reviewed a simialr PR 😄 

Anyway, the comments will be same as in https://github.com/huggingface/transformers/pull/41434, we do not add not-existent models to make auto-map happy. Instead we delete them from auto-map if they do not exist

omsherikar committed:
Merge branch 'main' into main

omsherikar commented:
Thanks for the guidance! @zucchini-nlp  I’ve updated the approach to follow the policy from #41434:
Reverted the alias.
Removed the non-existent perception_encoder from the auto-maps.
Kept the perception_lm entries intact.
Verified locally that MODEL_MAPPING.items() now succeeds.
I’ve opened a new PR with this change and will close this one. Let me know if you’d like me to fold it into this PR instead.

omsherikar committed:
Merge branch 'main' into main

omsherikar committed:
Merge branch 'main' into main

omsherikar committed:
Merge branch 'main' into main

omsherikar commented:
@zucchini-nlp  Could you please approve the workflows so CI can run? Happy to adjust anything else you’d like.

omsherikar committed:
Merge branch 'main' into main

omsherikar committed:
Merge branch 'main' into main
